lc2fen/infer_pieces.py

```diff
@@ -81,34 +81,34 @@ def __max_piece(tops):
     Returns the index of the piece with max probability.
     """
     # Set the initial maximum probability and index to the first piece in the list
-    value = tops[0][0][0]  # B(White Bishop)
+    value = tops[0][0]["B"]  # B(White Bishop)
```

Comment:
I don't think you can reference a position in an array like this. Have you tested it? I believe you should have to use the `__PREDS_DICT` here.

Reply:
With these changes you can also remove the comments, as the piece will already be indicated in the code.

Reply:
Hello @nayan2167 , why did you mark this as resolved? I can't see any changes/comment

Reply:
Hi @davidmallasen 
Apology  for Delay and for this mistake
I have made a few changes in **__max_pieces**, but I am getting an error while running test, I guess I am taking the wrong size of array (**pieces_probs**) in **infer_chess_pieces**, I tried to figure it out but I am not understanding it much, I need assistance for this.

![__max_piece](https://github.com/davidmallasen/LiveChess2FEN/assets/81795258/f27a11f6-7b1a-44c6-8749-ea80ffa63d51)
![test](https://github.com/davidmallasen/LiveChess2FEN/assets/81795258/7c93bad8-42e9-478a-a5f8-f6de069b5b4d)

Reply:
Hello @nayan2167 ,
No apologies needed, I'm glad that you were able to move forward! Now it looks good, but it still needs to be tested minimally. 
The error you are getting is because you are not generating the `pieces_probs_rand` correctly. The `pieces_probs` argument of `infer_chess_pieces` is a "List of the probabilities of each class in each position of the chessboard given in FEN notation order." This means that each of the 64 squares (the 64 positions in the list) is another list of probabilities of each chess piece belonging to that specific square. Maybe this should be clarified in the documentation (feel free to do so!).

I.e.
~~~
pieces_probs_rand = []
for _ in range(64):
    a = np.random.random(13)  # Each square will contain the probabilities of each of the 13 pieces
    a /= a.sum()  # The probabilities in each square should add up to 1
    pieces_probs_rand.append(a)
~~~

Let me know if this works for you.

Reply:
yup, it works!
get this output
```
Warning: the selected model is not accurate enough to predict a balanced board configuration
        Please consider providing the previous FEN, selecting a different model, or performing
        transfer learning on that model
```